cfme/cloud/flavor.py

```diff
@@ -210,6 +211,16 @@ def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True, c
         flavor = self.instantiate(name, provider, ram, vcpus, disk, swap, rxtx, is_public)
         return flavor
 
+    def all(self):
+        provider = self.filters.get('provider')  # None if no filter, need for entity instantiation
+        view = navigate_to(self, 'All')
+        result = []
+        for e in view.entities.get_all():
+            entity = self.instantiate(e.data['name'], provider or
+                                      get_crud_by_name(e.data['cloud_provider']))
```

Comment:
Required:
If you pass `OSP` with a `filter` it will return `ec2` entities with a provider as `OSP`. what do you think?